solr/contrib/ltr/src/java/org/apache/solr/ltr/ModelQuery.java

```diff
@@ -74,22 +74,29 @@
   protected Query originalQuery;
   // Original solr request
   protected SolrQueryRequest request;
+  protected LTRThreadModule ltrThreadMgr = null;
 
   public ModelQuery(LTRScoringModel ltrScoringModel) {
-    this(ltrScoringModel, Collections.emptyMap(), false);
+    this(ltrScoringModel, Collections.<String,String[]>emptyMap(), false, null);
   }
 
   public ModelQuery(LTRScoringModel ltrScoringModel, boolean extractAllFeatures) {
-    this(ltrScoringModel, Collections.emptyMap(), extractAllFeatures);
+    this(ltrScoringModel, Collections.<String, String[]>emptyMap(), extractAllFeatures, null);
   }
 
   public ModelQuery(LTRScoringModel ltrScoringModel, 
-      Map<String,String[]> externalFeatureInfo, 
-      boolean extractAllFeatures) {
+      Map<String, String[]> externalFeatureInfo, 
+      boolean extractAllFeatures, LTRThreadModule threadMgr) {
     this.ltrScoringModel = ltrScoringModel;
     this.efi = externalFeatureInfo;
-    this.extractAllFeatures = extractAllFeatures; 
-    querySemaphore = new Semaphore(LTRThreadModule.getMaxQueryThreads());
+    this.extractAllFeatures = extractAllFeatures;
+    if (threadMgr != null){
+       ltrThreadMgr = threadMgr;
+       querySemaphore = new Semaphore(ltrThreadMgr.getMaxQueryThreads());
```

Comment:
Do you need a semaphore if a non null manager is passed in with thread count <=1 ?